File: 4_step_4_binary_search/1_bs_on_1d_arrays/15_find_peak_element.cpp
#include <bits/stdc++.h>
using namespace std;

// brute force solution (Linear search)
int findPeakElement(vector<int> &arr);

int main()
{
    vector<int> arr = {1, 2, 3, 1};
    int result = findPeakElement(arr);
    cout << result;

    return 0;
}

int findPeakElement(vector<int> &arr)
{
    int n = arr.size();

    if (n == 1)
        return 0;

    if (arr[0] > arr[1])
        return 0;
    if (arr[n - 1] > arr[n - 2])
        return n - 1;

    for (int i = 1; i < n - 1; i++)
    {
        if (arr[i] > arr[i - 1] && arr[i] > arr[i + 1])
        {
            return i;
        }
    }

    return -1;
}

File: 4_step_4_binary_search/1_bs_on_1d_arrays/15_findPeakElement.js
// brute force solution

function findPeakElement(arr) {
    let n = arr.length

    if (n == 1 || arr[0] > arr[1]) return 0
    if (arr[n - 2] < arr[n - 1]) return n - 1;

    for (let i = 0; i < n - 1; i++) {
        if (arr[i - 1] < arr[i] && arr[i + 1] < arr[i]) {
            return i
        }
    }
    return -1;

}   

let arr = [1,2,4,5,2,3,1]
let result = findPeakElement(arr);
console.log(result)
